libs/translator/reader/read_geometry.cpp

```diff
@@ -400,7 +401,7 @@ void UsdArnoldReadMesh::Read(const UsdPrim &prim, UsdArnoldReaderContext &contex
     // we get smoothed normals by default.
     // Also, we only read the builting subdivisionScheme if the arnold
     // attribute wasn't explcitely set above, through primvars:arnold (see #679)
-    if ((!prim.HasAttribute(str::t_primvars_arnold_subdiv_type)) &&
+    if ((!HasAuthoredAttribute(prim, str::t_primvars_arnold_subdiv_type)) &&
```

Comment:
is it where it wasn't working ? on meshes ?

Reply:
yes this is what was causing test_0216 to produce the wrong results. This attribute "arnold:subdiv_type" isn't authored and we were supposed to enter the "if" condition and thus to set the proper subdivision attribute. But HasAttribute was always returning true since we introduced the schemas